### DISCUSSION TITLE
Clojars dependency vulnerabilities not reported

### DISCUSSION THREAD

**Discussion by mwing** [DependencyTrack/dependency-track#3159] [General]
Hi,

We have a Clojure project for which we created an SBOM with cdxgen. The SBOM imports fine to Dependency Track but it always shows 0 Audit Vulnerabilities. I tracked it down to the purl being of the type "pkg:clojars/javax.activation/activation@1.1", which is the analysis doesn't catch. However, if I just replace "clojars" with "maven" I start getting vulnerabilities. I have repo.clojars.org active under maven repository and I've also verified that it is active.

Is there any way I can enable the analysis without any string replacing? 

The components generated by cdxgen look like this:
`{
      "group" : "javax.activation",
      "name" : "activation",
      "version" : "1.1",
      "purl" : "pkg:clojars/javax.activation/activation@1.1",
      "type" : "library",
      "bom-ref" : "2bf8aab3-b444-460d-8f93-3761bc161516"
    }`

So if I change the compontents to:

`{
      "group" : "javax.activation",
      "name" : "activation",
      "version" : "1.1",
      "purl" : "pkg:maven/javax.activation/activation@1.1",
      "type" : "library",
      "bom-ref" : "2bf8aab3-b444-460d-8f93-3761bc161516"
    }`
    
Everything seems to work ok.

**Comment by nscuro** [comment]
@prabhu any reason for going with `clojars` in cdxgen? Isn't Clojure using Maven repositories behind the scenes?

**Reply by prabhu** [reply]
@nscuro I think clojars have their own repository with database and API. It is maven compatible but not maven.

https://github.com/clojars/infrastructure

**Comment by officerNordberg** [comment]
We use lein and maven to generate our sbom.  
```
lein update-in :pom-plugins conj '[org.cyclonedx/cyclonedx-maven-plugin "2.7.4" (:configuration [:projectType "library"] [:schemaVersion 1.3] [:includeLicenseText true] [:outputFormat json])]' -- pom  && mvn cyclonedx:makeBom
```
I believe this results in the matches you're looking for @mwing

**Reply by mwing** [reply]
Thanks, we'll test this out. However, we'd rather use a dockerised solution such as cdxgen as it's easier to include in all relevant builds.

**Reply by prabhu** [reply]
Have you tested with depscan? Not sure if it affects any other tools out there. 

https://github.com/owasp-dep-scan/dep-scan

**Reply by mwing** [reply]
@officerNordberg Unfortunately that approach produces the same results, ie. the purl contains "pkg:clojars/org.eclipse.jetty/jetty-http@9.4.28.v20200408", which Dependency Track doesn't match to a vulnerability. Changing it to "pkg:maven/org.eclipse.jetty/jetty-http@9.4.28.v20200408" does.

@prabhu dep-scan produces the same results as well unfortunately.

**Reply by officerNordberg** [reply]
Weird.  Our sboms have pkg:maven using the cmd I sent you.  I also syft the uber jar and merge the two to add cpes because I've had the worst accuracy with purls in general.  @see pkg:debian and now this.

```
    {
      "group" : "ring",
      "name" : "ring-jetty-adapter",
      "version" : "1.9.6",
      "description" : "Ring Jetty adapter.",
      "purl" : "pkg:maven/ring/ring-jetty-adapter@1.9.6?type=jar",
      "externalReferences" : [
        {
          "type" : "website",
          "url" : "https://github.com/ring-clojure/ring"
        },
        {
          "type" : "vcs",
          "url" : "https://github.com/ring-clojure/ring"
        }
      ],
      "type" : "library",
      "bom-ref" : "pkg:maven/ring/ring-jetty-adapter@1.9.6?type=jar"
    },
```

**Reply by officerNordberg** [reply]
I wonder if it's somehow because we are using a local maven proxy for clojars (as good supply chain hygiene dictates )  I can't see how that would cause that but I'll admit I don't know lein's pom creation process.

**Reply by mwing** [reply]
That could be it though, we'll have to investigate this a bit more.
